wallet/createtx.go

```diff
@@ -239,8 +250,14 @@ func (w *Wallet) findEligibleOutputs(dbtx walletdb.ReadTx, account uint32, minco
 		if err != nil || len(addrs) != 1 {
 			continue
 		}
-		_, addrAcct, err := w.Manager.AddrAccount(addrmgrNs, addrs[0])
-		if err != nil || addrAcct != account {
+		scopedMgr, addrAcct, err := w.Manager.AddrAccount(addrmgrNs, addrs[0])
+		if err != nil {
+			continue
+		}
+		if keyScope != nil && scopedMgr.Scope() != *keyScope {
+			continue
+		}
+		if addrAcct != account {
```

Comment:
Given that the internal account (maybe we should find another term for this to make it less confusing, accountID?) is unique, isn't this check enough, and we don't need the keyscope check above as well?

Reply:
They're not unique, each scoped key manager tracks accounts individually, which is why the key scope is required as well.